tests/testAutopas/Newton3OnOffTest.cpp

```diff
@@ -0,0 +1,122 @@
+
+  for (int i = 0; i < numMolecules; ++i) {
+    auto id = static_cast<unsigned long>(i);
+    autopas::Particle m(randomPosition(boxMin, boxMax), {0., 0., 0.}, id);
+    autoPas.addParticle(m);
+  }
+}
+
+TEST_F(Newton3OnOffTest, testAoS) {
+  for (auto containerOption : autopas::possibleContainerOptions) {
+    autoPas.init(getBoxMin(), getBoxMax(), getCutoff(), containerOption);
+    fillContainerWithMolecules(100, autoPas);
+
+    // with newton 3:
+    int callsNewton3 = 0;
+    EXPECT_CALL(mockFunctor, allowsNewton3()).WillOnce(Return(true));
+    EXPECT_CALL(mockFunctor, allowsNonNewton3()).WillOnce(Return(false));
+    EXPECT_CALL(mockFunctor, AoSFunctor(_, _, true))
+        .WillRepeatedly(testing::InvokeWithoutArgs([&]() { callsNewton3++; }));
+    autoPas.iteratePairwise(&mockFunctor, autopas::DataLayoutOption::aos);
+
+    // without newton 3:
+    int callsNonNewton3 = 0;
+    EXPECT_CALL(mockFunctor, allowsNewton3()).WillOnce(Return(false));
+    EXPECT_CALL(mockFunctor, allowsNonNewton3()).WillOnce(Return(true));
+    EXPECT_CALL(mockFunctor, AoSFunctor(_, _, true))
+        .Times(0);  // disables newton3 variant
+    EXPECT_CALL(mockFunctor, AoSFunctor(_, _, false))
+        .WillRepeatedly(
+            testing::InvokeWithoutArgs([&]() { callsNonNewton3++; }));
+    autoPas.iteratePairwise(&mockFunctor, autopas::DataLayoutOption::aos);
+
+    EXPECT_EQ(callsNewton3 * 2,
+              callsNonNewton3);  // should be called exactly two times
+  }
+}
+
+TEST_F(Newton3OnOffTest, testSoA) {
+  for (auto containerOption : autopas::possibleContainerOptions) {
+    autoPas.init(getBoxMin(), getBoxMax(), getCutoff(), containerOption);
+    fillContainerWithMolecules(100, autoPas);
+
+    // loader and extractor will be called, we don't care how often.
+    EXPECT_CALL(mockFunctor, SoALoader(_, _)).Times(testing::AtLeast(1));
+    EXPECT_CALL(mockFunctor, SoAExtractor(_, _)).Times(testing::AtLeast(1));
+
+    // with newton 3:
+    int callsNewton3SC = 0;    // same cell
+    int callsNewton3Pair = 0;  // pair of cells
+    EXPECT_CALL(mockFunctor, allowsNewton3()).WillOnce(Return(true));
+    EXPECT_CALL(mockFunctor, allowsNonNewton3()).WillOnce(Return(false));
+
+    // single cell
+    EXPECT_CALL(mockFunctor, SoAFunctor(_, true))
+        .WillRepeatedly(
+            testing::InvokeWithoutArgs([&]() { callsNewton3SC++; }));
+
+    // pair of cells
+    EXPECT_CALL(mockFunctor, SoAFunctor(_, _, true))
+        .WillRepeatedly(
+            testing::InvokeWithoutArgs([&]() { callsNewton3Pair++; }));
+
+    autoPas.iteratePairwise(&mockFunctor, autopas::DataLayoutOption::soa);
+
+    // without newton 3:
+    int callsNonNewton3SC = 0;
+    int callsNonNewton3Pair = 0;
+    EXPECT_CALL(mockFunctor, allowsNewton3()).WillOnce(Return(false));
+    EXPECT_CALL(mockFunctor, allowsNonNewton3()).WillOnce(Return(true));
+    EXPECT_CALL(mockFunctor, SoAFunctor(_, _, true))
+        .Times(0);  // disables newton3 variant
```

Comment:
This means we expect SoAFunctor with newton3 == true to be called zero times from here on?

Reply:
correct